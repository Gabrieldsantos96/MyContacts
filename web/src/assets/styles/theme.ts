export default {
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  },
  colors: {
    background: '#F6F5FC',
    primary: {
      lighter: '#E0E3FF',
      light: '#6674F4',
      main: '#5061FC',
      dark: '#3346F0'
    },
    warning: {
      light: '#FDA172',
      main: '#FC6A03',
      dark: '#DD571C'
    },
    danger: {
      light: '#F97171',
      main: '#FC5050',
      dark: '#F63131'
    },
    gray: {
      900: '#222222',
      200: '#BCBCBC',
      100: '#E5E5E5'
    },
    success: {
      main: '#51CA73'
    }
  },
  border: {
    radius: '0.4rem'
  },
  font: {
    family: 'Sora,sans-serif',
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '5.2rem'
    }
  }
} as const
