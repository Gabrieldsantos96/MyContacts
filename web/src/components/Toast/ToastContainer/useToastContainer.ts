import { useCallback, useState, useEffect } from 'react'
import { MessageProps } from '../ToastMessage'
import { IEventManager } from '@lib/eventManager'

export type EventManagerDataTransfer = Omit<MessageProps, 'id'>

export function useToastContainer({
  toastEventManager
}: {
  toastEventManager: IEventManager<EventManagerDataTransfer>
}) {
  const [messages, setMessages] = useState<MessageProps[]>([])

  function handleAddToast({ type, text, duration }: EventManagerDataTransfer) {
    setMessages((prevState: MessageProps[]) => [
      ...prevState,
      {
        id: Math.random(),
        type,
        text,
        duration
      }
    ])
  }

  useEffect(() => {
    toastEventManager.on('addToast', handleAddToast)

    return () => {
      toastEventManager.removeListener('addToast', handleAddToast)
    }
  }, [toastEventManager])

  const handleRemoveMessage = useCallback((id: number) => {
    setMessages((prevState) => prevState.filter((message) => message.id !== id))
  }, [])

  return {
    messages,
    handleRemoveMessage
  }
}
