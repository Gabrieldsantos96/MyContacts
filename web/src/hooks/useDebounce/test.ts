import { renderHook } from '@testing-library/react'
import { useDebounce } from '.'

vitest.useFakeTimers()

describe('useDebounce', () => {
  test('returns the initial value', () => {
    const { result } = renderHook(() => useDebounce({ value: 'initialValue' }))
    expect(result.current.debouncedValue).toBe('initialValue')
  })
})
