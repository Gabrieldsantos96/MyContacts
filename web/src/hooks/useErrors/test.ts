import { renderHook, act } from '@testing-library/react'
import useErrors from '.'

describe('useErrors', () => {
  test('initializes with no errors', () => {
    const { result } = renderHook(() => useErrors())

    expect(result.current.errors).toHaveLength(0)
    expect(result.current.noErrors).toBe(true)
  })

  test('adds and removes errors', () => {
    const { result } = renderHook(() => useErrors())

    const errorToAdd = { field: 'username', message: 'Username is required' }

    act(() => {
      result.current.setError([errorToAdd], 'username')
    })

    expect(result.current.errors).toEqual([errorToAdd])
    expect(result.current.noErrors).toBe(false)

    act(() => {
      result.current.removeError('username')
    })

    expect(result.current.errors).toHaveLength(0)
    expect(result.current.noErrors).toBe(true)
  })

  test('replaces existing errors for a field', () => {
    const { result } = renderHook(() => useErrors())

    const initialError = { field: 'email', message: 'Email is required' }
    const updatedError = { field: 'email', message: 'Invalid email format' }

    act(() => {
      result.current.setError([initialError], 'email')
    })

    expect(result.current.errors).toEqual([initialError])

    act(() => {
      result.current.setError([updatedError], 'email')
    })

    expect(result.current.errors).toEqual([updatedError])
  })

  test('gets error message by field name', () => {
    const { result } = renderHook(() => useErrors())

    const error = {
      field: 'password',
      message: 'Password must be at least 8 characters'
    }

    act(() => {
      result.current.setError([error], 'password')
    })

    const errorMessage = result.current.getErrorMessageByFieldName('password')

    expect(errorMessage).toBe(error.message)
  })
})
