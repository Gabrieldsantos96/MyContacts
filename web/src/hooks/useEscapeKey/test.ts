import { renderHook, act } from '@testing-library/react'
import { useEscapeKey } from '.'
import React from 'react'

describe('useModal', () => {
  it('should call when press escape', () => {
    const onCancelMock = vitest.fn()
    renderHook(() => useEscapeKey(onCancelMock))

    act(() => {
      const escapeEvent = new KeyboardEvent('keydown', { key: 'Escape' })
      document.body.dispatchEvent(escapeEvent)
    })

    expect(onCancelMock).toHaveBeenCalled()
  })

  it('should not call for other keys', () => {
    const onCancelMock = vitest.fn()

    renderHook(() => useEscapeKey(onCancelMock))

    act(() => {
      const otherKeyEvent = new KeyboardEvent('keydown', { key: 'Enter' })
      document.body.dispatchEvent(otherKeyEvent)
    })

    expect(onCancelMock).not.toHaveBeenCalled()
  })

  it('Should run useEffect cleanUp return function', () => {
    let isExecutedAfterUnmount = false

    vitest.spyOn(React, 'useEffect').mockImplementationOnce(() => {
      isExecutedAfterUnmount = true
    })

    const { unmount } = renderHook(() => useEscapeKey(vitest.fn()))

    unmount()

    expect(isExecutedAfterUnmount).toBe(true)
  })
})
