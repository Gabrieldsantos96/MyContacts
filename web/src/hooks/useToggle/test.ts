import { renderHook, act } from '@testing-library/react'
import { useToggle } from '.'

describe('useToggle', () => {
  test('initializes with default value', () => {
    const { result } = renderHook(() => useToggle(true))

    expect(result.current.value).toBe(true)
  })

  test('initializes with default value as false when not provided', () => {
    const { result } = renderHook(() => useToggle())

    expect(result.current.value).toBe(false)
  })

  test('toggles the value', () => {
    const { result } = renderHook(() => useToggle(true))

    act(() => {
      result.current.toggle()
    })

    expect(result.current.value).toBe(false)
  })

  test('sets the value directly', () => {
    const { result } = renderHook(() => useToggle(true))

    act(() => {
      result.current.setValue(false)
    })

    expect(result.current.value).toBe(false)
  })
})
