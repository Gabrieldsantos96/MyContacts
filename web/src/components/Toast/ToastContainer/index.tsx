import { useToastContainer } from './useToastContainer'
import { toastEventManager } from '@utils/helpers/toastEventManager/toastEventManager'
import ToastList from '../ToastList'

const ToastContainer = () => {
  const { handleRemoveMessage, messages } = useToastContainer({
    toastEventManager
  })
  return (
    <ToastList messages={messages} handleRemoveMessage={handleRemoveMessage} />
  )
}

export default ToastContainer
