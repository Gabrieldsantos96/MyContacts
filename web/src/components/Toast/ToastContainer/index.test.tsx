import { render } from '@utils/test/test-ui'

import ToastContainer from '.'
import theme from '@assets/styles/theme'

describe('<ToastContainer />', () => {
  it('should render toast container', () => {
    const { container } = render(<ToastContainer />)

    expect(container.firstChild).toHaveStyle({
      position: 'fixed',
      left: '50%',
      'z-index': 2,
      bottom: theme.spacings.xlarge,
      transform: 'translateX(-50%)'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
