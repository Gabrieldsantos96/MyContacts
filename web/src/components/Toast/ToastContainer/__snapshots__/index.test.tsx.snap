// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ToastContainer /> > should render toast container 1`] = `
<div
  class="sc-beySPh eUcDme"
/>
`;
